common/app/uiComponent/core/JavascriptRendering.scala

```diff
@@ -0,0 +1,105 @@
+package uiComponent.core
+
+import java.io._
+import java.nio.charset.StandardCharsets
+import javax.script.{CompiledScript, SimpleScriptContext}
+
+import common.Logging
+import model.ApplicationContext
+import play.api.Mode
+import play.api.libs.json.{JsValue, Json}
+import uiComponent.core.JavascriptEngine.EvalResult
+
+import scala.util.{Failure, Success, Try}
+
+trait JavascriptRendering extends Logging {
+
+  def javascriptFile: String
+
+  private implicit val scriptContext = createContext()
+  private val memoizedJs: Try[EvalResult] = loadJavascript()
+
+  def render(props: Option[JsValue] = None)(implicit ac: ApplicationContext): Try[String] = for {
+    propsObject <- encodeProps(props)
+    js <- if(ac.environment.mode == Mode.Dev) loadJavascript() else memoizedJs
```

Comment:
In prod the javascript bundle doesn't change so we can load it once.
When developing you want every rendering to take into account the lastest change made to the javascript bundle.

I can add a comment to explain it